spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDParser.java

```diff
@@ -43,4 +45,29 @@
 	 *         described by the DSL.
 	 */
 	List<ModuleDescriptor> parse(String name, String config, ParsingContext type);
+
+	public static class DefinitionOrException {
```

Comment:
This class should not exist. The FloController should simply handle an Exception being thrown from the repository.